openlibrary/core/ia.py

```diff
@@ -2,24 +2,258 @@
 """
 
 import logging
-import os
 import requests
+import simplejson
+import six
+import os
 import web
 
 from infogami import config
 from infogami.utils import stats
 
 from openlibrary.core import cache
+from openlibrary.core import helpers as h
 from openlibrary.utils.dateutil import date_n_days_ago
+from six.moves.urllib.parse import urlencode
 
-import six
 
 logger = logging.getLogger('openlibrary.ia')
 
 IA_BASE_URL = config.get('ia_base_url')
 VALID_READY_REPUB_STATES = ['4', '19', '20', '22']
 
 
+class IAEditionSearch:
+    """
+    Enables one to query archive.org editions, and return OL editions w availability.
+    """
+
+    MAX_EDITIONS_LIMIT = 20
+    AVAILABILITY_STATUS = 'loans__status__status'
+    VALID_SORTS = [
+        '__random', '__sort', 'addeddate', 'avg_rating',
+        'call_number', 'createdate', 'creatorSorter',
+        'creatorSorterRaw', 'date', 'downloads', 'foldoutcount',
+        'headerImage', 'identifier', 'identifierSorter', 'imagecount',
+        'indexdate', 'item_size', 'languageSorter', 'licenseurl',
+        'mediatype', 'mediatypeSorter', 'month', 'nav_order',
+        'num_reviews', 'programSorter', 'publicdate', 'reviewdate',
+        'stars', 'titleSorter', 'titleSorterRaw', 'week', 'year',
+        'loans__status__last_loan_date'
+    ]
+
+    PRESET_QUERIES = {
```

Comment:
A core library module seems like a strange place to store presets. I'd have thought they'd be in some calling code.